include/login_functions.php

```diff
@@ -127,46 +147,43 @@ function ldapAuthenticate($email, $password, $SessionCachePolicy, $rememberme)
-                    $sql = 'SELECT id,password FROM ' . qid('user') . " WHERE email='" . pdo_real_escape_string($email) . "'";
-                    $result = pdo_query("$sql");
+                    $user = new User();
+                    $userid = $user->GetIdFromEmail($email);
 
                     // If the user doesn't exist we add it
-                    if (pdo_num_rows($result) == 0) {
+                    if (!$userid) {
                         @$givenname = $entries[0]['cn'][0];
                         if (!isset($givenname)) {
                             $loginerror = 'No givenname (cn) set in LDAP, cannot register user into CDash';
                             return false;
                         }
                         $names = explode(' ', $givenname);
 
-                        $User = new User;
-
                         if (count($names) > 1) {
-                            $User->FirstName = $names[0];
-                            $User->LastName = $names[1];
+                            $user->FirstName = $names[0];
+                            $user->LastName = $names[1];
                             for ($i = 2; $i < count($names); $i++) {
-                                $User->LastName .= ' ' . $names[$i];
+                                $user->LastName .= ' ' . $names[$i];
                             }
                         } else {
-                            $User->LastName = $names[0];
+                            $user->LastName = $names[0];
                         }
 
                         // Add the user in the database
-                        $storedPassword = md5($password);
-                        $User->Email = $email;
-                        $User->Password = $storedPassword;
-                        $User->Save();
-                        $userid = $User->Id;
+                        $storedPassword = password_hash($password, PASSWORD_DEFAULT);
+                        $user->Email = $email;
+                        $user->Password = $storedPassword;
+                        $user->Save();
+                        $userid = $user->Id;
                     } else {
-                        $user_array = pdo_fetch_array($result);
-                        $storedPassword = $user_array['password'];
-                        $userid = $user_array['id'];
+                        $user->Id = $userid;
+                        $user->Fill();
 
                         // If the password has changed we update
-                        if ($storedPassword != md5($password)) {
-                            $User = new User;
-                            $User->Id = $userid;
-                            $User->SetPassword(md5($password));
+                        $passwordHash = password_hash($password, PASSWORD_DEFAULT);
+                        if ($user->Password != $passwordHash) {
+                            $user->Password = $passwordHash;
+                            $user->Save();
```

Comment:
`pdo_execute` would log any SQL error generated by the underlying UPDATE statement.